src/extension.ts

```diff
@@ -379,18 +391,51 @@ async function startServer() {
   let clientOptions: LanguageClientOptions = {
     documentSelector: [{ scheme: "file", language: "wdl" }],
     outputChannel: channel,
+    initializationFailedHandler: (error: ResponseError<InitializeError> | Error | any) => {
+      initializationError = error;
+      setStatus(Status.Error, "Failed to initialize Spocket language-service. Please check your configuration settings and reload this window.");
+      return false;
+    },
     errorHandler: {
       error: (error, message, count) => {
         return {
-          action: ErrorAction.Continue,
+          action: ErrorAction.Shutdown,
           message: `Sprocket encountered an error: ${error}`,
           handled: false,
         };
       },
-      closed: () => {
+      closed: async () => {
+        if (initializationError !== undefined) {
+          channel.appendLine("Initialization error: " + initializationError.message);
+          return {
+            action: CloseAction.DoNotRestart,
+            // This won't show an error dialog to the user,
+            // allowing us to display a custom error later.
+            message: "",
+          };
+        }
+
+        // We will retry atleast once if sprocket was initialized.
+        // Can we add a configuration property to allow for more retries?
```

Comment:
I think a future PR to add such a setting would be great.

Reply:
Opened an issue for it #25